code/HabitTracker/app/src/main/java/com/example/habittracker/DatabaseManager.java

```diff
@@ -343,18 +349,23 @@ public void getUserDetails(String username, UserDetailsCallback callback){
             @Override
             public void onComplete(@NonNull Task<DocumentSnapshot> task) {
                 if (task.isSuccessful()) {
-                    User user;
+                    HashMap<String,Object> userDetails = new HashMap<>();
                     DocumentSnapshot document = task.getResult();
                     if(!document.exists()){
                         callback.onCallbackFailed();
                         return;
                     }
-                    String hashedPassword = (String) document.getData().get("hashedPassword");
-                    user = new User(
-                            username,
-                            hashedPassword
-                    );
-                    callback.onCallbackSuccess(user);
+                    ArrayList<String> following = (ArrayList<String>) document.getData().get("following");
+                    ArrayList<String> followers = (ArrayList<String>) document.getData().get("followers");
+                    ArrayList<String> pendingFollowReqs = (ArrayList<String>) document.getData().get("pendingFollowReqs");
+                    ArrayList<String> pendingFollowerReqs = (ArrayList<String>) document.getData().get("pendingFollowReqs");
+                    userDetails.put("username",username);
+                    userDetails.put("following",following);
+                    userDetails.put("followers",followers);
+                    userDetails.put("pendingFollowReqs",pendingFollowReqs);
+                    userDetails.put("pendingFollowerReqs",pendingFollowerReqs);
```

Comment:
You're probably not adding the hashedPassword on purpose, correct?

Reply:
Yes, this is not for logging in, this is for the profile page, I assume we don't need the password, but I could add it?

Reply:
Yeah we shouldn't need the password. So this is good for now.